### PR TITLE
feat: move to cargo-xwin for clippy

### LINKED ISSUE #3507: Explore cargo-xwin for windows CI
Our windows CI is currently slow and inconsistent in performance. We could run e.g. the windows clippy job with cargo-xwin and see if we can speed up at least the feedback cycle on clippy:

```
cargo xwin clippy --target x86_64-pc-windows-msvc --workspace --all-targets --all-features --locked --profile fast-build -- -D warnings
```

### PR DESCRIPTION
**Pull request by samypr100**
## Summary

Move to cargo-xwin for clippy

Closes #3507

### CODE REVIEW

**Comment by messense** [comment]
FYI, consider install `cargo-xwin` from pypi to avoid compiling it. And you should try to cache Windows SDK download by explicitly setting `XWIN_CACHE_DIR` and cache it via the cache action.

**Comment by charliermarsh** [comment]
Would cargo-binstall work or no?

**Comment by messense** [comment]
probably works, I never tried.

**Comment by messense** [comment]
BTW, I think you forgot to use a Linux runner? otherwise using `cargo-xwin` on Windows isn't beneficial?

**Comment by samypr100** [comment]
Indeed, was trying to gather some baseline data on windows first 😅

**Comment by konstin** [comment]
The clippy step is 12:30min while clippy itself takes 30s, do you know where the other 12min are? The xwin download and unpack are <2min on my machine.

**Comment by messense** [comment]
> do you know where the other 12min are?

Probably because the network connection from GH action to Windows SDK download isn't very good,  cached version is much better.

**Review by konstin — APPROVED** [review]
Clippy itself takes 30s, the whole job 1:30min, great work!

**Comment by samypr100** [comment]
😅 need to add up the totals numbers and make a table, will do later today to get better comparison.

At a glance on an unchached run it's marginally worse by a quite the factor (2m-3m before vs 15 min now) ~ .
On cached runs job totals are actually about the same on both windows and linux.

TL;DR Seems when cache is busted the impact is much worse than it is now and we're not gaining on job total speeds.

**Comment by konstin** [comment]
Would moving the xwin cache into a separate cache step only keyed on `rust-toolchain.toml`? This way the cache would only ever be invalidated when we upgrade rust

**Comment by zanieb** [comment]
We do frequently fill our cache, we'll want to be careful that it's not evicted often due to space limitations.

**Comment by samypr100** [comment]
Here's some of the initial numbers to compare I've grabbing from all the past runs.

|  Current Main     | Before Cache (Step, Job)  | After Cache (Step, Job) |
|-------------------|---------------------------|-------------------------|
| Clippy on Windows | 2m~, 3m~                  | 30s~, 1.25m~            |

|  New Runs                         | After (Step, Job)
|-----------------------------------|------------------|
| xwin on windows                   | 2m 45s~, 5m~     |
| xwin on linux (no cache)          | 14m~, 15m~       |
| xwin on linux (rust cache)        | 12m~, 13m~       |
| xwin on linux (xwin cache)        | 2m~, 3m~         |
| xwin on linux (rust + xwin cache) | 30s~, 1.25m~     |

It seems clippy xwin runs about the same speed as windows currently (post the ReFS move). The job totals are aligned on best scenarios.

The xwin cache made the biggest difference for cargo xwin, making it run consistently with current windows runner, but it introduces a bigger risk when that cache gets busted.

The 15 minute hit will be most noticeable on downstream forks (where there is no cache often) or when cache is lost in main after rust toolchain upgrades or cache eviction.

**Comment by konstin** [comment]
Personally, i think this still is a noticeable improvement, it means we skip the virtual disk setup and have a windows runner less; the 15min cache priming about every 6 weeks are acceptable.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -72,48 +72,33 @@ jobs:
...
-          RUSTUP_HOME: ${{ env.DEV_DRIVE }}/.rustup
-        run: rustup component add clippy
-
-      - uses: Swatinem/rust-cache@v2
+        run: rustup target add x86_64-pc-windows-msvc
+      - name: Load xwin cache
+        uses: actions/cache@v4
+        with:
+          path: "${{ github.workspace}}/.xwin"
+          key: cargo-xwin-${{ hashFiles('rust-toolchain.toml') }}
```
Comment (messense) [inline .github/workflows/ci.yml:86]: I don't think xwin cache needs to be keyed by Rust version, it's just an unpacked Windows SDK, no Rust involved.

**Comment by samypr100** [comment]
> Personally, i think this still is a noticeable improvement, it means we skip the virtual disk setup and have a windows runner less; the 15min cache priming about every 6 weeks are acceptable.

Sounds good, I'll mark the PR ready

File: .github/workflows/ci.yml (hunk 1 above)
Comment (samypr100) [reply .github/workflows/ci.yml]: Thanks, moved cache key to be just `cargo-xwin`
